src/allmydata/test/cli/test_invite.py

```diff
@@ -0,0 +1,187 @@
+import os
+import mock
+import json
+from os.path import join
+
+from twisted.trial import unittest
+from twisted.internet import defer
+from ..common_util import run_cli
+from ..no_network import GridTestMixin
+from .common import CLITestMixin
+
+
+class _FakeWormhole(object):
+
+    def __init__(self, outgoing_messages):
+        self.messages = []
+        self._outgoing = outgoing_messages
+
+    def get_code(self):
+        return defer.succeed(u"6-alarmist-tuba")
+
+    def set_code(self, code):
+        self._code = code
+
+    def get_welcome(self):
+        return defer.succeed(
+            json.dumps({
+                u"welcome": {},
+            })
+        )
+
+    def allocate_code(self):
+        return None
+
+    def send_message(self, msg):
+        self.messages.append(msg)
+
+    def get_message(self):
+        return defer.succeed(self._outgoing.pop(0))
+
+    def close(self):
+        return defer.succeed(None)
+
+
+def _create_fake_wormhole(outgoing_messages):
+    return _FakeWormhole(outgoing_messages)
+
+
+class Join(GridTestMixin, CLITestMixin, unittest.TestCase):
+
+    @defer.inlineCallbacks
+    def setUp(self):
+        self.basedir = self.mktemp()
+        yield super(Join, self).setUp()
+        yield self.set_up_grid(oneshare=True)
+
+    @defer.inlineCallbacks
+    def test_create_node_join(self):
+        node_dir = self.mktemp()
+
+        with mock.patch('allmydata.scripts.create_node.wormhole') as w:
+            fake_wh = _create_fake_wormhole([
+                json.dumps({u"abilities": {u"server-v1": {}}}),
```

Comment:
missing, I think, is a test for the behavior when abilities does not contain `u"server-v1"`.